clang-tools-extra/include-cleaner/tool/IncludeCleaner.cpp

```diff
@@ -183,9 +195,26 @@ class Action : public clang::ASTFrontendAction {
     auto Results =
         analyze(AST.Roots, PP.MacroReferences, PP.Includes, &PI,
                 getCompilerInstance().getPreprocessor(), HeaderFilter);
-    if (!Insert)
+
+    if (!Insert) {
+      llvm::errs()
+          << "[WARNING] -insert is deprecated in favor of `-disable-insert`. "
+             "The old flag was confusing since it suggested that inserts "
+             "were disabled by default, when they were actually enabled. "
+             "See https://github.com/llvm/llvm-project/issues/132983\n";
```

Comment:
```suggestion
          << "warning: '-insert' is deprecated in favor of '-disable-insert'. "
             "The old flag was confusing since it suggested that inserts "
             "were disabled by default, when they were actually enabled.\n";
```